packages/Search/src/DTK_DistributedSearchTree_decl.hpp

```diff
@@ -0,0 +1,91 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#ifndef DTK_DISTRIBUTED_SEARCH_TREE_DECL_HPP
+#define DTK_DISTRIBUTED_SEARCH_TREE_DECL_HPP
+
+#include <Kokkos_View.hpp>
+
+#include <Teuchos_Comm.hpp>
+#include <Teuchos_RCP.hpp>
+
+#include <DTK_DBC.hpp>
+#include <DTK_LinearBVH.hpp>
+#include <details/DTK_DetailsDistributedSearchTreeImpl.hpp>
+
+#include "DTK_ConfigDefs.hpp"
+
+namespace DataTransferKit
+{
+
+template <typename DeviceType>
+class DistributedSearchTree
+{
+  public:
+    DistributedSearchTree(
+        Teuchos::RCP<Teuchos::Comm<int> const> comm,
+        Kokkos::View<Box const *, DeviceType> bounding_boxes );
+
+    template <typename Query>
+    void query( Kokkos::View<Query *, DeviceType> queries,
+                Kokkos::View<int *, DeviceType> &indices,
+                Kokkos::View<int *, DeviceType> &offset,
+                Kokkos::View<int *, DeviceType> &ranks ) const;
+
+    template <typename Query>
+    typename std::enable_if<
+        std::is_same<typename Query::Tag, Details::NearestPredicateTag>::value,
```

Comment:
I would say then if there is not an easy way to get them out (because they are not computed) then maybe we don't want this extra functionality here and instead will always compute the distances when we need them (which we always will).